src/main/java/org/jabref/gui/fieldeditors/LinkedFilesEditorViewModel.java

```diff
@@ -165,14 +166,25 @@ public void bindToEntry(BibEntry entry) {
         List<Path> newFiles = fileFinder.findAssociatedFiles(entry, dirs, extensions);
 
         List<LinkedFileViewModel> result = new ArrayList<>();
-        for (Path newFile : newFiles) {
-            boolean alreadyLinked = files.get().stream()
+        for (Path foundFile : newFiles) {
```

Comment:
This code now seems to have a huge overlap with the above changes. Can you please try to extract the common code?